flink-ml-lib/src/main/java/org/apache/flink/ml/feature/minmaxscaler/MinMaxScaler.java

```diff
@@ -0,0 +1,210 @@
+                output.collect(new StreamRecord<>(minVector));
+                output.collect(new StreamRecord<>(maxVector));
+            }
+        }
+
+        @Override
+        public void processElement(StreamRecord<DenseVector> streamRecord) {
+            DenseVector currentValue = streamRecord.getValue();
+            if (minVector == null) {
+                int vecSize = currentValue.size();
+                minVector = new DenseVector(vecSize);
+                maxVector = new DenseVector(vecSize);
+                System.arraycopy(currentValue.values, 0, minVector.values, 0, vecSize);
+                System.arraycopy(currentValue.values, 0, maxVector.values, 0, vecSize);
+            } else {
+                Preconditions.checkArgument(
+                        currentValue.size() == maxVector.size(),
+                        "CurrentValue should has same size with maxVector.");
+                for (int i = 0; i < currentValue.size(); ++i) {
+                    minVector.values[i] = Math.min(minVector.values[i], currentValue.values[i]);
+                    maxVector.values[i] = Math.max(maxVector.values[i], currentValue.values[i]);
+                }
+            }
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void initializeState(StateInitializationContext context) throws Exception {
+            super.initializeState(context);
+            minState =
+                    context.getOperatorStateStore()
+                            .getListState(
+                                    new ListStateDescriptor<>(
+                                            "minState",
+                                            getOperatorConfig()
+                                                    .getTypeSerializerIn(
+                                                            0, getClass().getClassLoader())));
+            maxState =
+                    context.getOperatorStateStore()
+                            .getListState(
+                                    new ListStateDescriptor<>(
+                                            "maxState",
+                                            getOperatorConfig()
+                                                    .getTypeSerializerIn(
+                                                            0, getClass().getClassLoader())));
+
+            OperatorStateUtils.getUniqueElement(minState, "minState").ifPresent(x -> minVector = x);
+            OperatorStateUtils.getUniqueElement(maxState, "maxState").ifPresent(x -> maxVector = x);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void snapshotState(StateSnapshotContext context) throws Exception {
+            super.snapshotState(context);
+            minState.clear();
+            maxState.clear();
+            if (minVector != null) {
+                minState.add(minVector);
+            }
+            if (maxVector != null) {
```

Comment:
nits: instead of using two `if` statements, would it be simpler to do the following?

```
if (minVector != null) {
    minState.add(minVector);
    maxState.add(maxVector);
}
```

Reply:
OK